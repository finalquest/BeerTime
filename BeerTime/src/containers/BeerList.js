import { connect } from 'react-redux';
import BeerListScreen from '../screens/BeerListScreen';
import getBeers from '../model/actions/getBeers';
import { GET_BEERS } from '../model/const/actionNames';

const mapStateToProps = ({ beer: { beers = [] }, fetching }) => {
  const refreshing = fetching[`${GET_BEERS}`];
  return {
    beers,
    refreshing,
  };
};
const mapDispatchToProps = dispatch => ({
  onMounted: () => dispatch(getBeers()),
});
export default connect(mapStateToProps, mapDispatchToProps)(BeerListScreen);
