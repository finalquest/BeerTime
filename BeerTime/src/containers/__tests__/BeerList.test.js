import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import BeerList from '../BeerList';

const mockStore = configureStore();
const initialState = {
  beer: {
    beers: [],
  },
  fetching: {},
};

describe('BeerList test suite', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <BeerList />,
      { context: { store: mockStore(initialState) } },
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should dispacth getBeer action on onMount', () => {
    const store = mockStore({ ...initialState });
    const wrapper = shallow(
      <BeerList />,
      { context: { store } },
    );
    wrapper.find('BeerListScreen').prop('onMounted')();
    expect(store.getActions()).toEqual(
      [{
        type: 'GET_BEERS',
        request: { url: 'https://api.punkapi.com/v2/beers' },
      }],
    );
  });
});
