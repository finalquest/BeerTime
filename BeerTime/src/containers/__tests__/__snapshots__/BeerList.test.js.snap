// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerList test suite renders as expected 1`] = `
<BeerListScreen
  beers={Array []}
  onMounted={[Function]}
  refreshing={false}
/>
`;
